### PR TITLE
Implement PUT request and handle headers

### PR DESCRIPTION
**Pull request by mindylou**
- Add in `PUT` request
- Handle `application/json` headers in general initializer

### CODE REVIEW

**Review by AAAstorga — APPROVED** [review]
LGTM!

**Review by kevinchan159 — APPROVED** [review]
lgtm!! 💯  i would also add the general init that we talked about

File: FutureNova/Classes/Endpoint.swift [hunk 1]
```diff
@@ -67,6 +75,35 @@ extension Endpoint {
...
+                            customHost: String? = nil,
+                            customScheme: String? = nil,
+                            customPort: Int? = nil) {
+        self.path = path
+        self.queryItems = queryItems
+
+        var modifiedHeaders = headers
+        if modifiedHeaders["Content-Type"] == nil {
+            // Set Content-Type to application/json so backend can identify json body
+            modifiedHeaders["Content-Type"] = "application/json"
```
Comment (kevinchan159) [inline FutureNova/Classes/Endpoint.swift]: don't think we need to set the `Content-Type` if there is no body so we can just keep it as 
`self.headers = headers`

**Review by kevinchan159 — APPROVED** [review]
✌️

**Pull request merged**